spec/rspec/support/method_signature_verifier_spec.rb

```diff
@@ -384,17 +384,21 @@ def arity_kw(x, y = {}, z:2); end
               expect(valid?(nil, :a => 1)).to eq(false)
             end
 
-            it 'treats symbols as keyword arguments and the rest as optional argument' do
-              expect(valid?(nil, 'a' => 1)).to eq(true)
-              expect(valid?(nil, 'a' => 1, :z => 3)).to eq(true)
-              expect(valid?(nil, 'a' => 1, :b => 3)).to eq(false)
-              expect(valid?(nil, 'a' => 1, :b => 2, :z => 3)).to eq(false)
+            it 'treats symbols as keyword arguments' do
+              expect(valid?(nil, :z => 3)).to eq(true)
+              expect(valid?(nil, :b => 3)).to eq(false)
+              expect(valid?(nil, :b => 2, :z => 3)).to eq(false)
+            end
+
+            it 'treats string keys as invalid keyword arguments' do
+              expect(valid?(nil, 'a' => 1)).to eq(false)
+              expect(valid?(nil, 'a' => 1, :z => 3)).to eq(false)
             end
```

Comment:
This needs to be changed to conditonally apply the new behaviour, the old spec was valid for Ruby 2.x

Typically we'd define whole different specs for different Rubies.

Reply:
Sure, added this as a RubyFeatures method and applied the change and the specs based on that flag.